crates/cli/src/subcommands/generate/mod.rs

```diff
@@ -157,41 +152,40 @@ pub async fn exec(config: Config, args: &clap::ArgMatches) -> anyhow::Result<()>
 
     format_files(paths.clone(), lang)?;
 
-    if delete_files {
-        let mut files_to_delete = vec![];
-        for entry in fs::read_dir(out_dir)? {
-            let entry = entry?;
-            let path = entry.path();
-            if path.is_file() {
-                if let Ok(contents) = fs::read_to_string(&path) {
-                    if !contents.starts_with("// THIS FILE IS AUTOMATICALLY GENERATED BY SPACETIMEDB.") {
-                        continue;
-                    }
-                }
-
-                if paths
-                    .iter()
-                    .any(|x| x.file_name().unwrap() == path.file_name().unwrap())
-                {
+    // TODO: We should probably just delete all generated files before we generate any, rather than selectively deleting some afterward.
```

Comment:
@bfops Yeah hold on I think there are some bugs here. We only prompt users *after* we've already done the generation? That seems wrong to me. Also we should only be replacing files which contain our special header, not just overwriting whatever file is already there. Can we fix those issues here or do you want to do a separate PR?

Reply:
Let's do those things in a separate PR since they're not API-breaking.
Also, it looks like we have the capacity to _generate_ files in subdirectories, but it's not clear to me that we can _delete_ them from subdirectories..

Reply:
Created https://github.com/clockworklabs/SpacetimeDB/issues/1786